### PR TITLE
Fix payments stream and test

### PR DESCRIPTION
**Pull request by asaf-erlich**
# Description of change
* Add note to payments.json schema
* Fix get_payments call in payments stream to pass in object type, not client

# Manual QA steps
 - tests passing again
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_square/streams.py [hunk 1]
```diff
@@ -98,9 +102,10 @@ class Payments():
     object_type = 'DISCOUNT'
 
     def sync(self, client, start_time, bookmarked_cursor): #pylint: disable=no-self-use
-        for page, cursor in client.get_payments(client, start_time, bookmarked_cursor):
+        for page, cursor in client.get_payments(self.object_type, start_time, bookmarked_cursor):
```
Comment (kspeer825) [inline tap_square/streams.py:105]: It is unclear to me what this does. The other stream classes pass in client here.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tap_square/streams.py (hunk 1 above)
Comment (asaf-erlich) [inline tap_square/streams.py:105]: Fixes the following error I saw when first running the test:
```
 2020-07-17 17:19:05,149Z    tap - Traceback (most recent call last):
        2020-07-17 17:19:05,149Z    tap -   File "/usr/local/share/virtualenvs/tap-square/bin/tap-square", line 33, in <module>
        2020-07-17 17:19:05,149Z    tap -     sys.exit(load_entry_point('tap-square', 'console_scripts', 'tap-square')())
        2020-07-17 17:19:05,149Z    tap -   File "/usr/local/share/virtualenvs/tap-square/lib/python3.5/site-packages/singer/utils.py", line 229, in wrapped
        2020-07-17 17:19:05,149Z    tap -     return fnc(*args, **kwargs)
        2020-07-17 17:19:05,149Z    tap -   File "/opt/code/tap-square/tap_square/__init__.py", line 20, in main
        2020-07-17 17:19:05,149Z    tap -     sync(args.config, args.state, catalog)
        2020-07-17 17:19:05,149Z    tap -   File "/opt/code/tap-square/tap_square/sync.py", line 38, in sync
        2020-07-17 17:19:05,149Z    tap -     for page, cursor in stream_obj.sync(client, start_time, bookmarked_cursor):
        2020-07-17 17:19:05,149Z    tap -   File "/opt/code/tap-square/tap_square/streams.py", line 101, in sync
        2020-07-17 17:19:05,149Z    tap -     for page, cursor in client.get_payments(client, start_time, bookmarked_cursor):
        2020-07-17 17:19:05,149Z    tap -   File "/opt/code/tap-square/tap_square/client.py", line 171, in get_payments
        2020-07-17 17:19:05,150Z    tap -     with singer.http_request_timer('GET ' + object_type):
        2020-07-17 17:19:05,150Z    tap - TypeError: Can't convert 'SquareClient' object to str implicitly
```

Comment (asaf-erlich) [reply tap_square/streams.py:105]: The input into that method is the object type. Also, the other classes pass in the object type so I think it's just a copy paste error.

Comment (asaf-erlich) [reply tap_square/streams.py:105]: For example: https://github.com/singer-io/tap-square/blob/c628bee4fe276e1d43aee2ad7fd4e0b76c2970da/tap_square/streams.py#L8

Comment (asaf-erlich) [reply tap_square/streams.py:105]: See my comment below which explains it and shows the error

**Review by luandy64 — APPROVED** [review]
The changes look good to me, but failing tests are not

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**